xbmc/peripherals/EventScanner.cpp

```diff
@@ -95,6 +95,10 @@ void CEventScanner::Process(void)
     const double nowMs = static_cast<double>(SystemClockMillis());
     const double scanIntervalMs = GetScanIntervalMs();
 
+    // Handle wrap-around
+    if (nowMs < nextScanMs)
+      nextScanMs = nowMs;
```

Comment:
You are right - I thought nowMs was being incremented. Looking closer the code looks okay.